packages/teleport/src/Welcome/NewCredentials/NewPasswordlessDevice.tsx

```diff
@@ -55,6 +55,8 @@ export function NewPasswordlessDevice(props: Props & SliderProps<LoginFlow>) {
     switchFlow('local', applyNextAnimation);
   }
 
+  const isFirefox = navigator?.userAgent?.toLowerCase().includes('firefox');
```

Comment:
TIL how the safe navigation operator actually works in JS. I've always thought that it's similar to `&.` in Ruby, so I thought the code would fail if `navigator.userAgent` resolves to undefined for some reason, this code is going to attempt to call `toLowerCase` on undefined, but it'll actually short-circuit just before that.

Though does `navigator?.` make sense actually? In case it's not defined in the browser for some reason, isn't the browser going to throw a ReferenceError anyway?

Perhaps we could do something like this:

```suggestion
  const isFirefox = window.navigator?.userAgent
    ?.toLowerCase()
    .includes('firefox');
```

Reply:
If you'd like to look up more about it, it's called the "optional chaining operator"